src/backofftimer.cpp

```diff
@@ -110,11 +111,16 @@ void BackoffTimer::update(dstime* waituntil)
 {
     if (next)
     {
+        assert(next != 1);
+        if (next == 1)
+        {
+            LOG_warn << "Possible wrong management of timer";
+        }
+
         if (next <= Waiter::ds)
         {
             *waituntil = 0;
             next = 1;
-            base = 1;
```

Comment:
The backoff shouldn't be reinitialized here because exponential backoffs don't work as expected. It's diffucult to reach this code in those circumstances, but it's possible.